frontend/www/js/omegaup/login/signin.ts

```diff
@@ -0,0 +1,89 @@
+import { OmegaUp } from '../omegaup';
+import { types } from '../api_types';
+import * as api from '../api';
+import * as ui from '../ui';
+import T from '../lang';
+import Vue from 'vue';
+import login_Signin from '../components/login/Signin.vue';
+import VueRecaptcha from 'vue-recaptcha';
+
+OmegaUp.on('ready', () => {
+  const payload = types.payloadParsers.LoginDetailsPayload();
+  new Vue({
+    el: '#main-container',
+    components: {
+      'omegaup-login-signin': login_Signin,
+      'vue-recaptcha': VueRecaptcha,
+    },
+    methods: {
+      loginAndredirect: (
+        usernameOrEmail: string,
+        password: string,
+        type: string,
+      ): void => {
+        api.User.login({
+          usernameOrEmail: usernameOrEmail,
+          password: password,
+        })
+          .then(() => {
+            const params = new URL(document.location.toString()).searchParams;
+            let pathname = params.get('redirect');
+            if (
+              !pathname ||
+              (pathname.indexOf('/') === 0 && type === 'register')
+            ) {
+              pathname = '/profile/';
+            } else if (
+              !pathname ||
+              (pathname.indexOf('/') === 0 && type === 'login')
+            ) {
```

Comment:
```suggestion
            } else if (pathname.indexOf('/') === 0 && type === 'login') {
```

si `pathname` es falsoso en L32, va a entrar al bloque en L35. eso quiere decir que no es posible que `pathname` sea falsoso aquí.

Reply:
Listo, creo que es mejor usar guard clause pattern, a ver si está bien así. Me quedó la duda si se requiere `return` después de cada `window.location`

Reply:
guard clause pattern es _aún_ mejor! sí, el `return` es necesario.